ocaml/xapi/xapi_session.ml

```diff
@@ -1195,6 +1196,13 @@ let local_logout ~__context =
   let session_id = Context.get_session_id __context in
   Xapi_local_session.destroy ~__context ~self:session_id
 
+let check_uname uname = 
+  match uname with
+  | "" ->
+    raise(Api_errors.Server_error
```

Comment:
is `failwith` better in this case as it is not due to server error, but less of argument ?

Reply:
@liulinC The exception `Api_errors.Server_error` is used for all API errors. Even `failwith`, which ends up raising a standard OCaml `Failure` exception, is eventually re-raised as a `Server_error` with `Api_errors.internal_error`.

The issue here is that this should not be an "internal error", because the server is not at fault here, but the client has given a bad value. In such cases, we want to raise a specific API error; see my other comment.